File: example/약수/약수의 합.py
import math
import sys
input = sys.stdin.readline

N = int(input())

# 1부터 N의 제곱근까지로 약수의 합 구하기


def divisor(n):
    sum = 0
    for i in range(1, int(math.sqrt(n))+1):
        if i*i == N:
            sum += i
        elif n % i == 0:
            sum += i+n//i
    return sum


print(divisor(N))

File: js/programmers/level1/약수의 합.js
function solution(n) {
  let sum = 0;
  for (let i = 1; i <= Math.sqrt(n); i++) {
    if (i ** 2 == n) sum += i;
    else if (n % i == 0) sum += i + n / i;
  }
  return sum;
}

console.log(solution(12));
console.log(solution(5));
